cmd/egress/cmd.go

```diff
@@ -99,3 +97,21 @@ are set correctly before execution.
 	return validateEgressCmd
 
 }
+
+func NewClient(ctx context.Context, logger ocmlog.Logger, config egressConfig) (cloudclient.CloudClient, error) {
```

Comment:
Why is the egress pkg implementing its own NewClient function? Shouldn't we make cloudclient handle this instead?

Reply:
It does handle the actual NewClient function. We can rename this to NewClientCaller or something. I separated this from the cmd following our previous comments on osdctl where I removed extra logic to a separate function to keep the cmd "Run" param simple. I can put it back up in "Run".. No preference.

Reply:
Forgot to add, I didn't do the same for dns and byovpc cmds, because I wasn't working on those 2 so didn't want to touch them too much other than preventing errors. Moving it back so they should all be the same.